### DISCUSSION TITLE
How to add an external library (zeromq.js) to the renderer? (electron+webpack+electron forge)

### DISCUSSION THREAD

**Discussion by devzzzero** [webpack/webpack#19013] [Q&A]
(Was #19001)

Hi, I am being hinted that https://github.com/zeromq/zeromq.js/issues/676 might be solvable with configuration changes for webpack.

In the repo, I have webpack.main.config.ts and webpack.renderer.config.ts

I set the following on the webpack.main.config.ts to get zeromq to load up properly on the main thread side.

  externals: {
    zeromq: 'zeromq',
    path: 'path',  // <- is this right?
    fs: 'fs',  // <- is this right?
    net: 'net', // <- is this right?
  },
How do I tell webpack to include fs, path, process, os node.js default packages for the renderer?
(just copying the above to webpack.renderer.ts over didn't work. )-:

Thank you @alexander-akait

Originally posted by @devzzzero in https://github.com/webpack/webpack/discussions/18995#discussioncomment-11350462

impacts this: https://github.com/electron/electron/issues/44805

**Comment by alexander-akait** [comment]
I have already given an answer earlier that you can't just use Node.js modules in `renderer`, `renderer` is a browser like place
